src/stdlib/SDL_getenv.c

```diff
@@ -18,6 +18,11 @@
      misrepresented as being the original software.
   3. This notice may not be removed or altered from any source distribution.
 */
+
+#if defined (__clang_analyzer__) && !defined(SDL_DISABLE_ANALYZE_MACROS)
```

Comment:
You can't commit this. It breaks precompiled header support. `#include "SDL_internal.h"` needs to be the first code in all source files.

Reply:
Maybe you can do this with CMake source file properties? @madebr?

Reply:
What :point_up: said, but use [`set_property`](https://cmake.org/cmake/help/latest/command/set_property.html#set-property) to append.
So something like:
```cmake
set_property(SOURCE
  src/stdlib/SDL_getenv.c
  src/stdlib/SDL_iconv.c
  src/stdlib/SDL_malloc.c
  src/stdlib/SDL_mslibc.c
  src/stdlib/SDL_qsort.c
  src/stdlib/stdlib.c
  src/stdlib/string.c
  src/stdlib/strtokr.c
    APPEND PROPERTY "SDL_DISABLE_ANALYZE_MACROS")
```

But this is not useful here because the macro is only needed when `__clang_analyzer__` is defined.
How about the following to construct `CLANG_TIDY_COMMAND`?
```cmake
# I leave it up to you to create nicer names :)
function(get_3rd_party_sources VAR)
    set(${VAR}
        "source1.c"
        ...
        "sourceN.c"
    PARENT_SCOPE)
endfunction()

function(get_clang_format_exceptions VAR)
    set(${VAR}
        "SDL_source1.c"
        ...
        "SDL_sourceN.c"
    PARENT_SCOPE)
endfunction()

function(construct_clang_filter_argument VAR SOURCES)
    set(res)
    foreach(src ${SOURCES})
        if(res)
            set(res "${res},")
        endif()
        set(res "${res}{\\\"name\\\":\\\"${src}\\\",\\\"lines\\\":[[1,1]]}")
    endforeach()
    set(${VAR} "[${res}]" PARENT_SCOPE)
endfunction()

get_3rd_party_sources(SOURCES_3RD_PARTY)
get_clang_format_exceptions(SOURCES_SDL_NO_FORMAT)
list(APPEND SOURCES_SDL_NO_FORMAT ${SOURCES_3RD_PARTY)

construct_clang_filter_argument(CLANG_LINE_FILTER ${SOURCES_SDL_NO_FORMAT})
set(CLANG_TIDY_COMMAND "${CLANG_TIDY_BINARY}" "-extra-arg=-Wno-unknown-warning-option" "--line-filter=${CLANG_LINE_FILTER}")
```